samples/NativeAOT/NativeAotTypeManager.cs

```diff
@@ -0,0 +1,206 @@
+using Java.Interop.Tools.TypeNameMappings;
+
+namespace NativeAOT;
+
+partial class NativeAotTypeManager : JniRuntime.JniTypeManager {
+
+	internal const DynamicallyAccessedMemberTypes Methods = DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods;
+	internal const DynamicallyAccessedMemberTypes MethodsAndPrivateNested = Methods | DynamicallyAccessedMemberTypes.NonPublicNestedTypes;
+
+	// TODO: list of types specific to this application
+	Dictionary<string, Type> typeMappings = new () {
+		["android/app/Activity"]                = typeof (Android.App.Activity),
+		["android/content/Context"]             = typeof (Android.Content.Context),
+		["android/content/ContextWrapper"]      = typeof (Android.Content.ContextWrapper),
+		["android/os/BaseBundle"]               = typeof (Android.OS.BaseBundle),
+		["android/os/Bundle"]                   = typeof (Android.OS.Bundle),
+		["android/view/ContextThemeWrapper"]    = typeof (Android.Views.ContextThemeWrapper),
+		["my/MainActivity"]                     = typeof (MainActivity),
+	};
+
+	public NativeAotTypeManager ()
+	{
+		AndroidLog.Print (AndroidLogLevel.Info, "NativeAotTypeManager", $"# jonp: NativeAotTypeManager()");
+
+		// TODO: should eventually be refactored in Mono.Android.dll
+		var jniToManaged = TypeManagerMapDictionaries.JniToManaged;
+		var managedToJni = TypeManagerMapDictionaries.ManagedToJni;
+		foreach (var pair in typeMappings) {
+			jniToManaged [pair.Key] = pair.Value;
+			managedToJni [pair.Value] = pair.Key;
+		}
+	}
+
+	public override void RegisterNativeMembers (
+			JniType nativeClass,
+			[DynamicallyAccessedMembers (MethodsAndPrivateNested)]
+			Type type,
+			ReadOnlySpan<char> methods)
+	{
+		AndroidLog.Print (AndroidLogLevel.Info, "NativeAotTypeManager", $"# jonp: RegisterNativeMembers: nativeClass={nativeClass} type=`{type}`");
+		
+		if (methods.IsEmpty) {
+			AndroidLog.Print (AndroidLogLevel.Info, "NativeAotTypeManager", "methods.IsEmpty");
+			return;
+		}
+
+		int methodCount = CountMethods (methods);
+		if (methodCount < 1) {
+			AndroidLog.Print (AndroidLogLevel.Info, "NativeAotTypeManager", $"methodCount < 1: {methodCount}");
+			return;
+		}
+
+		JniNativeMethodRegistration [] natives = new JniNativeMethodRegistration [methodCount];
+		int nativesIndex = 0;
+		MethodInfo []? typeMethods = null;
+
+		ReadOnlySpan<char> methodsSpan = methods;
+		bool needToRegisterNatives = false;
+
+		while (!methodsSpan.IsEmpty) {
+			int newLineIndex = methodsSpan.IndexOf ('\n');
+
+			ReadOnlySpan<char> methodLine = methodsSpan.Slice (0, newLineIndex != -1 ? newLineIndex : methodsSpan.Length);
+			if (!methodLine.IsEmpty) {
+				SplitMethodLine (methodLine,
+					out ReadOnlySpan<char> name,
+					out ReadOnlySpan<char> signature,
+					out ReadOnlySpan<char> callbackString,
+					out ReadOnlySpan<char> callbackDeclaringTypeString);
+
+				Delegate? callback = null;
+				if (callbackString.SequenceEqual ("__export__")) {
```

Comment:
This should error out/abort/etc., as `[Export]` is not (currently) supportable on NativeAOT.  (`jnimarshalmethod-gen` could do it!  If it fully worked!  Which it doesn't!)

This is basically dead code, and should be (mostly) removed.